src/tpm2_wrap.c

```diff
@@ -4421,6 +4421,42 @@ int wolfTPM2_NVReadAuth(WOLFTPM2_DEV* dev, WOLFTPM2_NV* nv,
     return rc;
 }
 
+int wolfTPM2_NVReadCert(WOLFTPM2_DEV* dev, TPM_HANDLE handle,
+    uint8_t* buffer, uint32_t* len)
+{
+    int rc;
+    WOLFTPM2_NV nv;
+    TPMS_NV_PUBLIC nvPublic;
+
+    XMEMSET(&nvPublic, 0, sizeof(nvPublic));
+    XMEMSET(&nv, 0, sizeof(nv));
+
+    /* Get or check size of NV */
+    rc = wolfTPM2_NVReadPublic(dev, handle, &nvPublic);
+    if (rc == 0) {
+        if (buffer == NULL) {
+            /* just return size */
+            *len = nvPublic.dataSize;
+            return 0;
+        }
+        if (nvPublic.dataSize > *len) {
+            return BUFFER_E;
+        }
```

Comment:
The wolfTPM2_NVReadCert API works for me with this additional change:
```
@@ -4442,6 +4442,8 @@ int wolfTPM2_NVReadCert(WOLFTPM2_DEV* dev, TPM_HANDLE handle,
         if (nvPublic.dataSize > *len) {
             return BUFFER_E;
         }
+
+        *len = nvPublic.dataSize;
     }
     else {
     #ifdef DEBUG_WOLFTPM
```

Otherwise len is potentially too large and get errors in wolfTPM2_NVReadAuth:
```
TPM2_NV_Read failed 326: TPM_RC_NV_RANGE: NV offset+size is out of range
```